core/src/main/java/feign/Capability.java

```diff
@@ -57,7 +57,7 @@ static Object enrich(Object componentToEnrich,
   static Object invoke(Object target, Capability capability, Class<?> capabilityToEnrich) {
     return Arrays.stream(capability.getClass().getMethods())
         .filter(method -> method.getName().equals("enrich"))
-        .filter(method -> method.getReturnType().isAssignableFrom(capabilityToEnrich))
+        .filter(method -> method.getReturnType() == capabilityToEnrich)
```

Comment:
I don't think this will work for Feign, all enrichers are based on interfaces....

I even wonder how this didn't broke any unit test.

Reply:
> I don't think this will work for Feign, all enrichers are based on interfaces....
> 
> I even wonder how this didn't broke any unit test.

I think you're right, if you look at the feign application as a whole, this rarely happens, since I'm thinking about the logic of this place in terms of this method, thank you, I'll close this pull request,